test/distributed/_tensor/test_dtensor_compile.py

```diff
@@ -270,6 +281,49 @@ def test_2d_fsdp_tp_compile(self):
 
         self.assertEqual(out, compiled_output)
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    def test_2d_fsdp_tp_ac_compile(self):
+        dp_degree = 2
+        tp_degree = self.world_size // dp_degree
+        model = SimpleModel(self.device_type)
+        model_copy = copy.deepcopy(model)
+        enable_2d_with_fsdp()
+
+        # 2-D mesh is [dp, tp]
+        mesh_2d = init_device_mesh(
+            "cuda", mesh_shape=(dp_degree, tp_degree), mesh_dim_names=("dp", "tp")
+        )
+
+        inp = torch.rand(20, 10, device=self.device_type)
+        parallelize_plan = {
+            "mlp_0.net1": ColwiseParallel(),
+            "mlp_0.net2": RowwiseParallel(),
+            "mlp_1.net1": ColwiseParallel(),
+            "mlp_1.net2": RowwiseParallel(),
+        }
+        tp_model = parallelize_module(model, mesh_2d["tp"], parallelize_plan)
+        tp_model = checkpoint_wrapper(
+            tp_model,
+            checkpoint_impl=CheckpointImpl.NO_REENTRANT,
+            checkpoint_fn=checkpoint,
+            use_reentrant=False,
+        )
```

Comment:
Just one more point here, with TP AC, users usually use `input_reshard` together.

Reply:
hmmm I don't understand in what case user would use `input_reshard`, we don't use it at least

Reply:
double checked on this `input_reshard`, you mean for TP only (not SP) this would reduce the memory usage when combining with FSDP checkpoint wrapper? wondering why we didn't apply this to our models when applying TP, we should probably add it. Is it because baseline don't have this feature?